src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -0,0 +1,631 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "Components/PipelineExecutionEstimatorImpl.h"
+#include "Components/TrainingOnlyEstimatorImpl.h"
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wold-style-cast"
+#   pragma clang diagnostic ignored "-Wsign-conversion"
+#   pragma clang diagnostic ignored "-Wdocumentation-unknown-command"
+#   pragma clang diagnostic ignored "-Wzero-as-null-pointer-constant"
+#   pragma clang diagnostic ignored "-Wextra-semi-stmt"
+#   pragma clang diagnostic ignored "-Wmissing-noreturn"
+#   pragma clang diagnostic ignored "-Wdocumentation"
+#   pragma clang diagnostic ignored "-Wdouble-promotion"
+#   pragma clang diagnostic ignored "-Wcast-align"
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#   pragma clang diagnostic ignored "-Wshadow"
+#elif (defined _MSC_VER)
+#   pragma warning(push)
+#   pragma warning(disable: 4127)
+#endif
+
+#include "../3rdParty/eigen/Eigen/Dense"
+
+#if (defined __clang__)
+#   pragma clang diagnostic pop
+#elif (defined _MSC_VER)
+#   pragma warning(pop)
+#endif
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+static constexpr char const * const         SVDComponentsEstimatorName("SVDComponentsEstimator");
```

Comment:
>static [](start = 0, length = 6)

static means local to the current CPP file. When including this you are creating multiple definitions of the same string in multiple CPP files that are not related to your implementation. Please, put this into a CPP file and declare here an extern.

Reply:
This is a constexpr value, which requires immediate assignment. Perhaps the "static" can be removed?